graphql-api.cabal

```diff
@@ -35,11 +35,11 @@ library
     , graphql
     , text
   exposed-modules:
-      GraphQL.Definitions
       GraphQL.Input
       GraphQL.Output
       GraphQL.Schema
-      GraphQL.TypeApi
+      GraphQL.TypedApi
+      GraphQL.TypedSchema
```

Comment:
Glad you like most of these names. Do you want to do that rename in this PR? (I think you should, but am also happy to see a different PR).

Re `Value`, I think they sort of are literals, because they are derived from http://hackage.haskell.org/package/graphql-0.3/docs/Data-GraphQL-AST.html#g:4 but with `Variable` taken away. To put another way, a constructed `Value` cannot be resolved any further. It is terminal. I guess we could call it `Terminal` or `Terms`.

Leave `Value` as-is for now until we figure out something we both like.